### PR TITLE
SymbolExternalizer: Refactor Get_Range_Of_Identifier

### PR DESCRIPTION
**Pull request by marcosps**
Now we use the same function to return ranges on specific macros and on general sources.

### CODE REVIEW

File: libcextract/SymbolExternalizer.cpp [hunk 1]
```diff
@@ -37,10 +37,12 @@
    the identifier.  */
 #define TOKEN_VECTOR " ().,;+-*/^|&{}[]<>^&|\r\n\t"
 
-static std::vector<SourceRange>
-Get_Range_Of_Identifier_In_SrcRange(const SourceRange &range, const char *id)
+/* Return the ranges for all identifiers on the ids vector */
+template <typename T>
+static std::vector<std::pair<std::string, SourceRange>>
+Get_Range_Of_Identifier(const SourceRange &range, T ids)
```
Comment (giulianobelinassi) [inline libcextract/SymbolExternalizer.cpp:43]: It should be `T &ids`, or maybe `const T &ids`. The way it is written you will copy the datastructure, which is not needed.

File: libcextract/SymbolExternalizer.cpp [hunk 2]
```diff
@@ -72,10 +74,11 @@ Get_Range_Of_Identifier_In_SrcRange(const SourceRange &range, const char *id)
   return ret;
 }
 
-static std::vector<SourceRange>
-Get_Range_Of_Identifier_In_SrcRange(const SourceRange &range, const StringRef id)
+static std::vector<std::pair<std::string, SourceRange>>
+Get_Range_Of_Identifier(const SourceRange &range, const StringRef id)
```
Comment (giulianobelinassi) [inline libcextract/SymbolExternalizer.cpp:78]: `const StringRef &id`

**Review by giulianobelinassi — CHANGES_REQUESTED** [review]
Looks nice! you should only be careful with the references here.

File: libcextract/SymbolExternalizer.cpp (hunk 1 above)
Comment (marcosps) [reply libcextract/SymbolExternalizer.cpp]: done

File: libcextract/SymbolExternalizer.cpp (hunk 2 above)
Comment (marcosps) [reply libcextract/SymbolExternalizer.cpp]: done

**Comment by marcosps** [comment]
the new version contains both improvements

**Review by giulianobelinassi — APPROVED** [review]

**Pull request merged**